src/Aperture/elliptical.jl

```diff
@@ -0,0 +1,100 @@
+export EllipticalAperture,
+       EllipticalAnnulus
+
+"""
+   EllipticalAperture(x, y, a, b, theta)
+
+An elliptical aperture.
+
+Where a >= b > 0 and theta in degrees
+
+# Examples
+```jldoctest
+julia> EllipticalAperture(2,2,2,2,1.2)
+EllipticalAperture(2.0, 2.0, a=2.0, b=2.0, theta=1.2)
+
+```
+"""
+struct EllipticalAperture{T <: Number} <: AbstractAperture
+    x::T
+    y::T
+    a::T
+    b::T
+    theta::T
+end
+
+function EllipticalAperture(x, y, a, b, theta)
+    if (b < 0.0 || a < b )
+            error("Invalid ellipse parameters. Require a >= b > 0.0")
+    end
+    theta = mod(theta, 360)
+    return EllipticalAperture(promote(x, y, a, b, theta)...)
+end
+
+function Base.show(io::IO, e::EllipticalAperture)
+    print(io, "EllipticalAperture($(e.x), $(e.y), a=$(e.a), b=$(e.b), theta=$(e.theta)°)")
+end
+
+function oblique_coefficients(a, b, theta)
+    sintheta, costheta = sincos(deg2rad(theta))
```

Comment:
I know I suggested this change, but as Mosé pointed out we can just use 
`sincosd(theta)`

Reply:
I checked the documentation for sincosd, it is supported only in v1.3 and above. What practise should I follow so that this runs amongst different versions of Julia?

Reply:
ah, I see. To keep support for v1.0 revert to using `sincos(deg2rad())`

Reply:
Yes, for this specific function it's probably best to keep this way

Reply:
In the v1.0 Julia supports functions like tand and cotd which can be used here. So should I use these functions? Although it breaks the uniformity in code yet the performance gets better.